model/metrics/dotnet-metrics.yaml

```diff
@@ -0,0 +1,278 @@
+groups:
+  - id: metric.dotnet.process.cpu.count
+    type: metric
+    metric_name: dotnet.process.cpu.count
+    brief: "The number of processors available to the process."
+    note: >
+      Meter name: `System.Runtime`; Added in: .NET 9.0.
+
+      This metric reports the same values as accessing [`Environment.ProcessorCount`](https://learn.microsoft.com/dotnet/api/system.environment.processorcount).
+    instrument: updowncounter
+    unit: "{cpu}"
+    stability: experimental
+
+  - id: metric.dotnet.process.cpu.time
+    type: metric
+    metric_name: dotnet.process.cpu.time
+    brief: "CPU time used by the process as reported by the CLR."
+    note: >
+      Meter name: `System.Runtime`; Added in: .NET 9.0.
+
+      This metric reports the same values as accessing the corresponding
+      processor time properties on [`System.Diagnostics.Process`](https://learn.microsoft.com/dotnet/api/system.diagnostics.process).
+    instrument: counter
+    unit: "s"
+    stability: experimental
+    attributes:
+      - ref: cpu.mode
+        requirement_level: required
+
+  - id: metric.dotnet.process.memory.working_set
+    type: metric
+    metric_name: dotnet.process.memory.working_set
+    brief: "The number of bytes of physical memory mapped to the process context."
+    note: >
+      Meter name: `System.Runtime`; Added in: .NET 9.0.
+
+      This metric reports the same values as calling [`Environment.WorkingSet`](https://learn.microsoft.com/dotnet/api/system.environment.workingset).
+    instrument: updowncounter
+    unit: "By"
+    stability: experimental
+
+  - id: metric.dotnet.gc.collections
+    type: metric
+    metric_name: dotnet.gc.collections
+    brief: "The number of garbage collections that have occurred since the process has started."
+    note: >
+      Meter name: `System.Runtime`; Added in: .NET 9.0.
+
+      This metric reports the same values as calling [`GC.CollectionCount(int generation)`](https://learn.microsoft.com/dotnet/api/system.gc.collectioncount).
```

Comment:
@stevejgordon not sure if you saw this comment, could you please take a look?

Reply:
@noahfalk / @lmolkova - I've pushed a change making this 

```
This metric uses the [`GC.CollectionCount(int generation)`] https://learn.microsoft.com/dotnet/api/system.gc.collectioncount) API to calculate exclusive collections per generation.
```

How does that sound to you?

Reply:
I'd still be worried that the simple (but wrong) interpretation will be to assume generation N of this metric is `GC.CollectionCount(N)`. I know you used 'exclusive' trying to signal there is a difference but I doubt many people realize that GC.CollectionCount() API isn't exclusive already. I'd suggest we either describe the inclusive vs. exclusive distinction in more detail or don't mention GC.CollectionCount() at all.

Reply:
Perhaps, in this case, we don't overcomplicate things by mentioning the API used then since the implementation is more nuanced. That way we can just be clear on what the metric represents, but not how it does it.